src/GameLogic/PlugIns/GoldenInvasionPlugIn.cs

```diff
@@ -0,0 +1,390 @@
+    {
+        public DateTime NextRunUtc { get; set; } = DateTime.UtcNow;
+
+        public MobsInvasionState State { get; set; } = MobsInvasionState.NotStarted;
+
+        public ushort MapId { get; set; } = LorenciaId;
+
+        public readonly IGameContext Context;
+
+        public GameMapDefinition Map => this.Context.Configuration.Maps.First(m => m.Number == this.MapId);
+
+        public string MapName => this.Map.Name;
+
+        public GameServerState(IGameContext ctx)
+        {
+            this.Context = ctx;
+        }
+    }
+
+    private static readonly PeriodicInvasionConfiguration DefaultConfiguration = new()
+    {
+        EventDuration = 300000,
+        PreStartMessageDelay = 3000,
+        IsActive = true,
+        Message = "[{mapName}] Golden Invasion!",
+        Timetable = new(GeneratePeriodsSequence(TimeSpan.FromMinutes(1))),
+    };
+
+    private static readonly ushort LorenciaId = 0;
+    private static readonly ushort DeviasId = 2;
+    private static readonly ushort NoriaId = 3;
+    private static readonly ushort AtlansId = 7;
+    private static readonly ushort TarkanId = 8;
+
+    private static readonly ushort GoldenBudgeDragonId = 43;
+    private static readonly ushort GoldenGoblinId = 78;
+    private static readonly ushort GoldenSoldierId = 54;
+    private static readonly ushort GoldenTitanId = 53;
+    private static readonly ushort GoldenDragonId = 79;
+    private static readonly ushort GoldenVeparId = 81;
+    private static readonly ushort GoldenLizardKingId = 80;
+    private static readonly ushort GoldenWheelId = 83; // ?
+    private static readonly ushort GoldenTantallosId = 82;
+
+    private static readonly Random _random = new();
+
+    private static readonly ushort[] PossibleMaps = new[] { LorenciaId, NoriaId, DeviasId };
+
+    private static readonly Dictionary<ushort, ushort> GeneralMobsCountOnSelectedMap = new()
+    {
+         { GoldenDragonId, 10 },
+    };
+
+    private static readonly Dictionary<ushort, Dictionary<ushort, ushort>> UniqueMobsCountByMapId = new()
+    {
+        {
+            LorenciaId,
+            new()
+            {
+                { GoldenBudgeDragonId, 20 },
+            }
+        },
+        {
+            NoriaId,
+            new()
+            {
+                { GoldenGoblinId, 20 },
+            }
+        },
+        {
+            DeviasId,
+            new()
+            {
+                { GoldenSoldierId, 20 },
+                { GoldenTitanId, 10 },
+            }
+        },
+        {
+            AtlansId,
+            new()
+            {
+                { GoldenVeparId, 20 },
+                { GoldenLizardKingId, 10 },
+            }
+        },
+        {
+            TarkanId,
+            new()
+            {
+                { GoldenWheelId, 20 },
+                { GoldenTantallosId, 10 },
+            }
+        },
+    };
+
+    private static readonly ConcurrentDictionary<IGameContext, GameServerState> _states = new();
```

Comment:
Right now it uses because each GS/interfaces has own instance of this plugin. So, we create only one goldenInvasionState per GS.